### ISSUE TITLE
plugin import issues with latest dev build on windows os

### ISSUE THREAD

**Issue by ryantxu** [grafana/grafana#9541]
I have an plugin app with panels that use `import` to load other js files.  This used to work fine, but in the latest master, it does not

My file structure looks like:
```
src
├── module.js
├── panels
│   ├── agent-panel
│   │   ├── editor.html
│   │   ├── module.html
│   │   ├── module.js    <<<< the file I'm working on
│   │   └── plugin.json
│   ├── base-natel-panel.js <<<< file I want to import
│   ├── more-panels
│   │   ├── ...
├── datasource
│   ├── datasource.js
│   ├── ...
└── plugin.json
```

From `panels/agent-panel/module.js` my import is:
```
import {BaseNatelPanelCtrl} from  '../base-natel-panel';
```
This used to work fine, but now it tries to load the file from:
/public/plugins/grafana-natel-app/base-natel-panel.js

When I change the import to:
```
import {BaseNatelPanelCtrl} from  '../panels/base-natel-panel';
```
It loads properly.

I don't think that is the expected behavior, but let me know if it is :)

**Comment by torkelo** [comment]
Strange, maybe caused by systemjs upgrade.

**Comment by ryantxu** [comment]
Here is another weird one.  From my plugin, i have:
```
import {Emitter} from 'app/core/utils/emitter';
```
That is not found.  I tried changing to 
```
import {Emitter} from 'app/core/utils/emitter.ts';
```

The error I get is:
> Error: Unable to dynamically transpile ES module A loader plugin needs to be configured via `SystemJS.config({ transpiler: 'transpiler-module' })`. Instantiating http://localhost:3000/public/app/core/utils/emitter.ts Loading http://localhost:3000/public/plugins/grafana-natel-app/datasource/agent.js Loading plugins/grafana-natel-app/datasource/module

**Comment by ryantxu** [comment]
Ah on Emitter, I see the note in `PLUGIN_DEV.md`  can we add Emitter to the exposed resources

**Comment by ryantxu** [comment]
More testing here... the plugin structure works fine on the linux build, but does not work on windows.  I tested with:

https://ci.appveyor.com/project/Torkeldegaard/grafana/build/5106

perhaps some windows bug with webpack root?

**Comment by torkelo** [comment]
strange. So loading using relative paths does not work when you run grafana-server on windows?

**Comment by ryantxu** [comment]
correct. Using relative path in my plugin:
```
import {BaseNatelPanelCtrl} from  '../base-natel-panel';
```
 * works fine on linux/osx
 * resolves to an invalid path on windows

I tested using the windows build from your appveyor account

**Comment by ryantxu** [comment]
I can work around this by not using relative imports.

**Comment by daniellee** [comment]
I cannot recreate this on Windows - seems to work fine with the latest master build. @ryantxu can you confirm that you are still seeing this issue?

**Comment by ryantxu** [comment]
I still get an error when I do:
```
import {XXX} from '../panels/YYY'
```

But it works when I prefix a `./`
```
import {XXX} from './../panels/YYY'
```

both versions work on linux/osx.  This is an easy enough workaround, so i will close the issue.

Thanks!

**Comment by ryantxu** [comment]
Turns out this is not really solved ☹️  

The root cause is that paths with a backslash are getting passed into `importPluginModule` and then I assume the relative path resolution gets funky.

If i normalize the path in [plugin_loader.ts#L128](https://github.com/grafana/grafana/blob/master/public/app/features/plugins/plugin_loader.ts#L128)
```
-  return System.import(path);
+  return System.import(path.replace('\\', '/'));
```
Then everything works OK.

I created a [sample application](https://github.com/NatelEnergy/sample-grafana-app) that will duplicate the problem.  Try adding `Panel A`, and you will see:
```
Plugin component error Error: Fetch error: 404 Not Found
  Instantiating http://localhost:3000/public/plugins/sample-grafana-app/base-panel.js
  Loading http://localhost:3000/public/plugins/sample-grafana-app/panels\example-a-panel/module.js
  Loading plugins/sample-grafana-app/panels\example-a-panel/module
    at :3000/public/build/0.9dc160b3bd1f18eacada.js:1
    at <anonymous>
```

There may be a better fix -- i'm not sure how the backslash is generated in the first place!  Note the path with a backslash loads OK, but the relative paths from there are broken.

**Comment by daniellee** [comment]
@ryantxu thanks for taking the time to create a sample app with the problem. Will test it out on Windows.

**Comment by ryantxu** [comment]
Not sure if this helps, but another place I see a backslash on windows where I would expect a forward slash... in a custom datasource ConfigCtrl, I put this in the html:
```
<pre>{{ ctrl | json }}</pre>
```

And see this module path in the result:

```
{
  "meta": {
    "name": "Natel Agent: View",
    "type": "datasource",
    "id": "natel-agent-view",
    "enabled": false,
    "pinned": false,
    "module": "plugins/grafana-natel-app/datasource\\view/module",
    "baseUrl": "public/plugins/grafana-natel-app",
    "info": {
      "author": {
...
```

**Comment by daniellee** [comment]
I could recreate this on Windows. See if I get a fix done for this tomorrow.

**Comment by daniellee** [comment]
Fixed by ccf1a5124b53493c05c1d1e856f2d3d45cc5d3ce

**Comment by ryantxu** [comment]
thanks!
